flang/lib/Lower/Bridge.cpp

```diff
@@ -395,7 +396,26 @@ class FirConverter : public Fortran::lower::AbstractConverter {
                                fir::getTypeParams(hexv));
           return fir::substBase(hexv, temp);
         });
-    return bindSymbol(sym, exval);
+
+    bool bindStatus = bindSymbol(sym, exv);
+    if (bindStatus && init) {
+      if (auto seqTy = symType.dyn_cast<fir::SequenceType>()) {
+        Fortran::lower::StatementContext stmtCtx;
+        createSomeArrayAssignment(*this, exv, hexv, localSymbols, stmtCtx);
+        stmtCtx.finalize();
+      } else {
+	if (hsb.toExtendedValue().getBoxOf<fir::MutableBoxValue>()) {
```

Comment:
Should this be combined into an `else if (...)`? There is something strange with the indentation?

Reply:
Done.